core/src/test/scala/org/apache/spark/eventhubs/utils/RetryUtilsSuite.scala

```diff
@@ -17,8 +17,73 @@
 
 package org.apache.spark.eventhubs.utils
 
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+
+import com.microsoft.azure.eventhubs.EventHubException
 import org.scalatest.FunSuite
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class RetryUtilsSuite extends FunSuite with ScalaFutures {
+
+  def failedWithEHE: Future[Int] = Future.failed(new EventHubException(true, "failedWith"))
```

Comment:
If you want to write ideomatic Scala you should:
- Convert the def to a call
- Split the constants to a companion object